From 5c708d0b33d534951884b638853d604e94575e53 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Tue, 14 Jan 2020 20:49:45 +0200
Subject: [PATCH] [SW-8228] arcboot: add efuse to arcboot_rom

check debug interface disable fuse and call security
function to disable it if true

Change-Id: I0aed07f3d08b2350fa1a9ca857d5aaffa04491ae
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index c4759f3..33d5001 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -6,9 +6,15 @@
  */
 
 #include "arcboot.h"
+#include "efuse.h"
+#include "security.h"
 
 void arcboot_rom_init()
 {
+	u32 config_fuse = 0;
+	fuse_read(0x3A,&config_fuse);
+	if (config_fuse & BIT(12))
+		security_disable_debug_if();
 
 }
 
